UnitySDK/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -138,10 +168,41 @@ public void LazyInitialization()
             }
         }
 
+        public void EnableAutomaticStepping()
+        {
+            if (m_Stepper != null)
+            {
+                return;
+            }
+
+            m_StepperObject = new GameObject("AcademyStepper");
+            // Don't show this object in the hierarchy
+            m_StepperObject.hideFlags = HideFlags.HideInHierarchy;
+            m_StepperObject.AddComponent<AcademyStepper>();
+            m_Stepper = m_StepperObject.GetComponent<AcademyStepper>();
+        }
+
+        public void DisableAutomaticStepping()
+        {
+            if (m_Stepper == null)
+            {
+                return;
+            }
+
+            m_Stepper = null;
+            UnityEngine.Object.Destroy(m_StepperObject);
```

Comment:
@surfnerd Is this right, or do I need to Destroy the MonoBehaviour too?

Reply:
Should the AcademyStepper object destroy itself when the Academy disappears?
You should explicitly call `using UnityEngine;` on top of file so we know we rely on it in this script.

Reply:
> Should the AcademyStepper object destroy itself when the Academy disappears?

Yeah, I was missing that. Changing so that the Academy deletes it in Dispose()

> You should explicitly call using UnityEngine; on top of file so we know we rely on it in this script

`using UnityEngine` is already at the top. I think it's necessary here to disambiguate between https://docs.microsoft.com/en-us/dotnet/api/system.object?view=netframework-4.8

Reply:
Unit tests were complaining about using Destroy instead of DestroyImmediate in Editor mode. I changed things around a bit to use DestroyImmediate from Dispose (but will still use Destroy when disabling normally)